src/jackfreqd.c

```diff
@@ -763,7 +763,7 @@ void drop_privileges(char *setgid_group, char *setuid_user) {
 		char xdgDir[32];
 
 		sprintf(xdgDir, "/run/user/%s", setuid_user);
-		if (setenv("XDG_RUNTIME_DIR", xdgDir, 1))
+		if (setenv("XDG_RUNTIME_DIR", xdgDir, 0))
```

Comment:
This change looks as wring: if XDG_RUNTIME_DIR has already set in a desktop environment, but jackd is running by another user, jackfreqd will not able to connect properly.

Unfortunally, you haven't written what problem are you resolving.

Reply:
My issue was that I'm not using systemd, instead setting XDG_RUNTIME_DIR in a subdirectory of /tmp. /run/user is a systemd convention that users of other init systems might not be adhering to. This fix got it working for me but I do see your point, not sure what the best way of handling both cases would be.

Reply:
May be a better solution is to allow to specify the XDG_RUNTIME_DIR or it's base with a configuration parameter?